crates/subspace-farmer/src/dsn.rs

```diff
@@ -154,47 +153,40 @@ impl DSNSync for subspace_networking::Node {
 
         // obtain closest peers to the middle of the range
         let key = Code::Identity.digest(&middle);
-        let peers = self
-            .get_closest_peers(key)
-            .await
-            .map_err(GetPiecesByRangeError::CannotGetClosestPeers)?;
-
-        // select appropriate peer for the piece-by-range protocol
-        if peers.is_empty() {
-            return Err(GetPiecesByRangeError::NoClosestPiecesFound);
-        }
-
-        let mut peer_id: Option<PeerId> = None;
-        for peer_id_candidate in peers {
-            let result = self
-                .send_generic_request(peer_id_candidate, PeerInfoRequest)
-                .await;
-
-            trace!(
-                peer_id = %peer_id_candidate,
-                "Peer info request returned {:?}",
-                result
-            );
+        let peer_id = {
+            let node = self.clone();
+
+            backoff::future::retry(
+                backoff::ExponentialBackoffBuilder::new()
+                    .with_max_elapsed_time(Some(std::time::Duration::from_secs(1)))
```

Comment:
This timeout should be placed as a constant and IMHO be increased. I understood it as `the whole backoff-retry block time` and I would set it to 3-7 seconds.